<h1 style="
    font-family: Garamond;
    font-size: 34px;
    color: #2774AE;
    font-weight: 700;
    text-shadow:
        -1px -1px 0 #000,
         1px -1px 0 #000,
        -1px  1px 0 #000,
         1px  1px 0 #000;
">
Replication Report on <em>"A Five-Factor Asset Pricing Model"</em> (Fama & French, 2015)
</h1>

<h2 style="
    font-family: Garamond;
    font-size: 22px;
    color: #FFD100;
    font-style: italic;
    text-shadow:
        -1px -1px 0 #000,
         1px -1px 0 #000,
        -1px  1px 0 #000,
         1px  1px 0 #000;
">
Econ 430
</h2>

<h3 style="
    font-family: Garamond;
    font-size: 20px;
    color: #2774AE;
    font-style: italic;
">
Jacob Williams, Josh Kentworthy, and Ignacio Ramirez
</h3>

<h3 style="
    font-family: Garamond;
    font-size: 20px;
    color: #000000;
    font-style: italic;
">
University of California, Los Angeles
</h3>

<h3 style="
    font-family: Garamond;
    font-size: 20px;
    color: #000000;
    font-style: italic;
">
November 20, 2025
</h3>


<h3 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.4em;">
Original Paper
</h3>

<p style="font-family: Georgia; font-size: 15px; line-height: 1.5;">

<strong>Fama, Eugene F., &amp; French, Kenneth R.</strong><br>
<em>A Five-Factor Asset Pricing Model.</em><br>
<em>Journal of Financial Economics</em>, <strong>116</strong>(1), 1–22. (2015)

</p>


<h1 style="font-family: Garamond; font-weight: bold; font-size: 28px; margin-bottom: 0.4em;">
1. Original Study Summary
</h1>

<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
(a) Research Question: Clearly state the paper’s main research question and economic motivation
</h2>


Modern asset-pricing began with a simple premise: if markets are competitive and information flows freely, prices should adjust to news and leave no predictable profit opportunities. This idea is what became the **Efficient Market Hypothesis (EMH)**. EMH emerged in the 1950s and 60s out of early empirical work showing that stock returns had almost no autocorrelation, suggesting prices behave like a fair game responding only to new information.

A decade later, Sharpe (1964) and Lintner (1965) formalized the first equilibrium model linking risk to expected returns: the **Capital Asset Pricing Model (CAPM)**. Its prediction was:

<h4 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.2em;">
(1) CAPM Formula
</h4>


$$
E[R_i] - R_f = \beta_i\,(E[R_M] - R_f)
$$

CAPM originally told us that one thing explains expected returns: **market risk**. 

With **beta** as the sole predictor of cross-sectional differences in average returns. For 20 years this model remained empirically reasonable.

---

But as Fama repeatedly emphasizes, CAPM’s “golden age” was short. By the early 1980s and into the 1990s, empirical papers accumulated showing that the relationship between market betas and average returns was basically "flat", but according to CAPM it couldn't be flat, average returns should increase with market beta. So beta alone could not explain average returns sorted on firm **size** (Banz 1981), **earnings-price ratios** (Basu 1977), **leverage** (Bhandari 1988), or **book-to-market** (Rosenberg, Reif & Lanstein 1985).

These accumulating anomalies eventually shifted professional consensus. In a 1992 New York Times article based on a phone interview with Eugene Fama, he stated that “beta as the sole variable explaining returns on stocks is dead” (Berg 1992).

These empirical failures exposed a deeper theoretical problem: CAPM offered too narrow a representation of risk. Asset pricing models required additional dimensions to explain why some portfolios earned persistently higher returns than others.

When Eugene Fama and Kenneth French aimed at capturing these missing dimensions, they developed the **Fama–French Three-Factor Model (1993)**, which looked at the two biggest anomalies for the CAPM, which was it's inability to explain small stock returns and the difference between value and growth stock returns. That's why in the Three Factor Model they augmented the market factor with size (SMB: small minus big) and value (HML: high minus low book-to-market):
  
<h4 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.2em;">
(2) Fama-French Three Factor Formula
</h4>

$$
R_i - R_f = \alpha_i + \beta_i (R_M - R_f) + s_i\,\text{SMB} + h_i\,\text{HML} + \varepsilon_i
$$

<h4 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.2em;">
Research Question of the Three-Factor Model
</h4>

The central question motivating the Three-Factor Model is whether **size (SMB)** and **value (HML)** contain systematic information about expected returns that the CAPM’s single beta cannot capture. In other words:

**Can the cross section of average stock returns be better explained by adding size and value factors to the market factor, thereby accounting for the persistent return premiums observed for small firms and high book to market firms?**

After the Three-Factor Model (1993), Fama and French showed that the size and value factors **SMB** and **HML** captured important patterns that the CAPM could not explain. Small firms earned higher average returns than big firms (**size premium**), and high book to market firms earned higher returns than growth firms (**value premium**). In the three-factor framework, these effects appear through the parameters $( s_i )$ and $( h_i )$:

 $( s_i > 0 )$ for small firms and $( s_i < 0 )$ for big firms  
 $( h_i > 0 )$ for high book-to-market (value) firms

But the model was still incomplete as evident by Novy-Marx (2012), Titman, Wei and Xie (2004) which say that the Three Factor is incomplete model for expected returns because its three factors miss much of the variation in average returns related to profitability and investment. 

---
A natural next step after the Three-Factor Model was to ask whether basic valuation logic could justify adding additional predictors. Under the standard dividend-discount framework,

$
m_t = \sum_{k=1}^{\infty} \frac{E_t[d_{t+k}]}{(1+r)^k}
$

**holding other characteristics constant**, a firm with higher expected profitability should have a higher required return $( r )$. Put differently, if two firms have similar book equity $( B )$ but one has higher expected earnings or cash flows, the valuation identity implies a positive link between profitability and expected returns. Empirically, portfolios sorted on operating profitability $(\text{OP})$ showed exactly this pattern.

Likewise, investment behavior carries information about expected returns. In standard steady state valuation, higher investment $( I_t )$ relative to assets $( A_t )$ (i.e., high asset growth) is typically associated with lower expected returns because firms invest more when their marginal $( q )$ is low. This motivates a negative relation between investment and expected returns. Fama and French operationalize this using the investment factor:

$
\text{CMA} = R_{\text{Conservative}} - R_{\text{Aggressive}}
$

Combining theis economic motivation, Fama and French extend the Three-Factor Model by adding profitability and investment factors:

<h4 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.2em;">
(3) Fama-French Five Factor Formula
</h4>

$$
R_{i,t} - R_{f,t}
= \alpha_i
+ \beta_i \left(R_{M,t} - R_{f,t}\right)
+ s_i\,\text{SMB}_t
+ h_i\,\text{HML}_t
+ r_i\,\text{RMW}_t
+ c_i\,\text{CMA}_t
+ \varepsilon_{i,t}.
+ $$
<h4 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.2em;">
Research Question of the Five-Factor Model
</h4>

The Five-Factor Model asks whether **profitability (RMW)** and **investment (CMA)**, two characteristics central to standard valuation theory, further improve our understanding of the cross-section of expected returns beyond size and value. Meaning:

**Do differences in firm's profitability and investment behavior help explain average returns in a way that is consistent with basic discounted cash flow logic, and do these two factors capture return patterns not absorbed by the original three factors?**

The Five-Factor Model tests whether expected returns vary **positively with profitability** and **negatively with investment**, completing the move from anomaly driven factors toward factors grounded in economic fundamentals.

In this Five Factor Model, the economic prediction and what the data confirm is that **average returns vary positively with profitability** $(r_i > 0)$ **and negatively with investment** $(c_i < 0)$.

<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
(b) Theoretical Framework: Summarize the economic theory underlying the analysis
</h2>

<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
0. CAPM
</h2>

The Capital Asset Pricing Model (CAPM) predicts that expected excess returns depend only on market risk:

$E[R_i - R_f] = \beta_i \, E[R_M - R_f].$

Empirically, we estimate:

$R_{i,t} - R_{f,t}
= \alpha_i + \beta_i (R_{M,t} - R_{f,t}) + \varepsilon_{i,t}.$


**CAPM said that market beta *$\beta$* should fully explain expected returns**, leaving no room for abnormal performance *$\alpha$*.  
In theory, **$\alpha$** is the extra return beyond what your risk exposure predicts, the “bonus” return, but it shouldn’t exist if the model is good at explaining returns.

However, when researchers tested CAPM, they found return patterns, especially for small firms and high book-to-market firms that produced large, persistent **abnormal returns** (*$\alpha$*).  
These anomalies revealed that CAPM was missing important sources of risk, motivating the move to the **Three-Factor Model**.


---

<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
1. Fama–French Three-Factor Model
</h2>

<p style="text-align: center; font-family: Garamond; font-weight: bold; font-size: 20px;">
3 Factor Regression
</p>

$$
R_{i,t} - R_{f,t}
= \alpha_i
+ b_i (R_{M,t} - R_{f,t})
+ s_i \, \text{SMB}_t
+ h_i \, \text{HML}_t
+ \varepsilon_{i,t}.
$$

**Outcome variable:**  
$(R_{i,t} - R_{f,t})$: excess return on portfolio $i$ at time $t$.

**X Variables:**

**Market $(R_M - R_f)$**  
Represents aggregate systematic risk (the original CAPM beta risk).

**SMB (Size)**  
Small firms tend to have more volatile and cyclical earnings, weaker balance sheets, and greater sensitivity to recessions.  
Higher exposure to SMB reflects greater “small-firm risk,” so portfolios with larger SMB loadings should earn higher expected returns:

$s_i \uparrow \;\Rightarrow\; E[R_i] \uparrow$

**HML (Value)**  
High Book-to-Market (value) firms typically have low prices relative to fundamentals, persistently weak earnings, and characteristics associated with financial distress.  
Low Book-to-Market (growth) firms tend to be more profitable and less distressed.

If markets price risk rationally, portfolios with higher HML loadings are more exposed to “value/distress” risk and therefore earn higher expected returns:

$h_i \uparrow \;\Rightarrow\; E[R_i] \uparrow$

The three-factor model is thus a **risk-based extension of CAPM** that absorbs the size and value patterns left unexplained by beta alone.

---

<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
2. Fama–French Five-Factor Model
</h2>

<p style="text-align: center; font-family: Garamond; font-weight: bold; font-size: 20px;">
5 Factor Regression
</p>

$$
R_{i,t} - R_{f,t}
= \alpha_i
+ b_i (R_{M,t} - R_{f,t})
+ s_i \, \text{SMB}_t
+ h_i \, \text{HML}_t
+ r_i \, \text{RMW}_t
+ c_i \, \text{CMA}_t
+ \varepsilon_{i,t}.
$$

**Outcome variable:**  
$(R_{i,t} - R_{f,t})$: excess return on portfolio $i$ at time $t$.

**X Variables:**

Market $(R_M - R_f)$ – aggregate systematic risk.  

SMB – small-minus-big size factor.  

HML – high-minus-low value factor.  

**RMW – Robust minus Weak profitability.**  

**CMA – Conservative minus Aggressive investment.**

---

<h3 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
Valuation Economic Interpretation
</h3>

The Five-Factor Model is built directly from a simple valuation identity: a stock’s price must equal the discounted value of its future cashflows. In the dividend discount model,

$$
m_t = \sum_{\tau=1}^{\infty} \frac{E(d_{t+\tau})}{(1+r)^\tau},
$$

the price *must* reflect expectations of future dividends and the discount rate $r$, the stock’s long-run expected return. If two firms are expected to pay the same future dividends but one is cheaper today, the cheaper firm must have a higher expected return. This is the basic insight behind why valuation ratios like B/M forecast returns.

Miller and Modigliani (1961) show that the same logic applies to total equity, where cashflows can be decomposed into earnings minus investment:

$$
M_t = \sum_{\tau=1}^{\infty} 
\frac{E(Y_{t+\tau} - dB_{t+\tau})}{(1+r)^\tau},
$$

$$
\frac{M_t}{B_t}
=
\sum_{\tau=1}^{\infty}
\frac{E(Y_{t+\tau} - dB_{t+\tau})}{B_t (1+r)^\tau}.
$$

The third equation tells us that expected returns depend on three ingredients:

**(1) Valuation:** higher $B_t/M_t$ (cheap stocks) imply higher expected returns.  

**(2) Profitability:** higher expected earnings $E(Y_{t+\tau})$ imply a higher expected return.  

**(3) Investment:** higher expected growth in book equity $dB_{t+\tau}$ implies a lower expected return, because investing heavily today raises future earnings but lowers near-term cashflows available to shareholders.

The economic challenge, historically, was finding real-world proxies for expected profitability and expected investment. Novy-Marx (2013) shows that operating profitability strongly predicts average returns, and Aharoni, Grundy, and Zeng (2013) show that investment does as well. These patterns are only weakly captured by the original three-factor model.

Fama and French’s key insight is that RMW and CMA are the *natural* empirical proxies implied by the valuation decomposition: RMW captures differences in expected profitability, and CMA captures differences in expected investment. Together, they extend the three-factor model so that it lines up with the fundamental accounting identity linking prices, earnings, and investment.

---

**Profitability (RMW)**  

Higher expected earnings raise the numerator of the valuation identity.  
To keep the observed price fixed, the discount rate (expected return) must rise:

$E(Y_{t+\tau}) \uparrow \;\Rightarrow\; r \uparrow$

Portfolios with higher RMW loadings are more exposed to profitability risk and should earn higher expected returns:

$r_i \uparrow \;\Rightarrow\; E[R_i] \uparrow$

**Investment (CMA)**  

Higher investment increases future book equity, reducing net cashflow $(Y_{t+\tau} - dB_{t+\tau})$ for a given price.  
This pushes the required return **down**:

$dB_{t+\tau} \uparrow \;\Rightarrow\; r \downarrow$

Portfolios with higher CMA loadings behave like firms that invest conservatively (low $dB$) and thus carry higher expected returns than aggressively investing firms:

$c_i \uparrow \;\Rightarrow\; E[R_i] \downarrow$


<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
(c) Methodology: Econometric Approach, Model Specification, and Estimation
</h2>

The econometric approach follows the standard Fama–French asset pricing methodology. On the right-hand side (RHS), each model uses a set of traded factor returns, market excess return for CAPM. Market, SMB, and HML for the Three-Factor Model; and market, SMB, HML, RMW, and CMA for the Five-Factor Model to explain variation in portfolio excess returns. We estimate factor loadings (aka the time-series regression slopes) using time-series OLS regressions of portfolio excess returns on factor returns, and we evaluate model performance through the main diagnostics used in Fama and French (1993, 2015): the magnitude and significance of intercepts, the signs and t-statistics of factor loadings, the regression $R^2$ values that indicate how much common variation the factors capture, and the joint GRS test of whether all alphas are zero. We also compare models using the variance of the estimated alphas $alpha_i$, and examine whether adding profitability and investment factors reduces the magnitude and cross-sectional variance of these pricing errors across the 25 portfolio and 32 portfolio sorts.

We estimate factor loadings using time-series Ordinary Least Squares (OLS) regressions of portfolio excess returns on factor returns. For each test portfolio $i$, we run a separate regression of monthly excess returns $(R_{it} - R_{Ft})$ on the factor returns. This produces the intercept $(a_i)$ and the slope coefficients $(b_i, s_i, h_i, r_i, c_i)$, which measure the portfolio’s exposure to the market, size, value, profitability, and investment factors.

Our left side (LHS) portfolios follow the Fama and French (2015) design. Portfolios are formed from independent sorts on Size, Book-to-Market (B/M), Operating Profitability (OP), and Investment (Inv), and are then used as LHS assets in the regressions:

- 25 Size–B/M portfolios (5 x 5 sort on Size and B/M)  
- 25 Size–OP portfolios (5 x 5 sort on Size and operating profitability)  
- 25 Size–Inv portfolios (5 x 5 sort on Size and investment)  
- 32 Size–B/M–OP portfolios (2 x 4 x 4 sort on Size, B/M, and OP)  
- 32 Size–B/M–Inv portfolios (2 x 4 x 4 sort on Size, B/M, and investment)  
- 32 Size–OP–Inv portfolios (2 x 4 x 4 sort on Size, OP, and investment)

For each of these six 'left side' sets, we estimate the same sequence of time-series regressions on the right side (x) factors:

- CAPM: market excess return only  
- Three-Factor Model: market, SMB, HML  
- Five-Factor Model: market, SMB, HML, RMW, CMA  

This means, for example, that the 25 Size–B/M portfolios generate 25 separate CAPM regressions, 25 three-factor regressions, and 25 five-factor regressions, and similarly for each of the other 25 portfolio and 32 portfolio sorts. In total, this yields hundreds of time-series regressions, as in Fama and French (2015).

In the original three-factor paper (Fama and French, 1993), the same time-series OLS framework is applied to 25 Size–B/M stock portfolios and 7 bond portfolios. In our replication, we focus on the equity portfolios and estimate the three-factor model and the five-factor model on the same sets of stock portfolios, so model comparisons are based on identical LHS assets.

<h3 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.3em;">
Model Specification
</h3>

The Three-Factor Model is estimated using:

$$
R_{it} - R_{Ft}
= a_i
+ b_i (R_{Mt} - R_{Ft})
+ s_i \, SMB_t
+ h_i \, HML_t
+ e_{it}.
$$

For the Three-Factor Model, we follow the original Fama and French (1993) design and estimate the model **only** on the 25 Size–B/M portfolios. These portfolios are constructed from a 5 x 5 independent sort on Size and Book-to-Market and were the primary equity test assets used in the original three-factor paper. 

Thus, the Three-Factor Model produces **25 separate time-series regressions**, one for each Size–B/M portfolio.

<br>

The Five-Factor Model extends the Three-Factor Model by including profitability and investment:

$$
R_{it} - R_{Ft}
= a_i
+ b_i (R_{Mt} - R_{Ft})
+ s_i \, SMB_t
+ h_i \, HML_t
+ r_i \, RMW_t
+ c_i \, CMA_t
+ e_{it}.
$$

Once we move to the Five-Factor Model, we expand the left-hand-side (LHS) assets to match the full testing design of Fama and French (2015). Profitability and investment sorts did not exist in the earlier three-factor model, so they enter the analysis only at this stage.

The Five-Factor Model is estimated on the six portfolio families introduced in Fama and French (2015):

- 25 Size–B/M portfolios  
- 25 Size–OP portfolios  
- 25 Size–Inv portfolios  
- 32 Size–B/M–OP portfolios  
- 32 Size–B/M–Inv portfolios  
- 32 Size–OP–Inv portfolios  

Each portfolio in each sort generates one time-series regression. This produces **171 separate regressions** per model specification and allows direct comparison of the Three-Factor and Five-Factor Models on a greatly expanded set of characteristics.

This structure mirrors the original papers: the Three-Factor Model is evaluated on the traditional 25 Size–B/M portfolios, while the Five-Factor Model is evaluated on the complete set of 25-portfolio and 32-portfolio sorts that incorporate profitability and investment information.


The coefficients $(b_i, s_i, h_i, r_i, c_i)$ represent exposures to the underlying economic risks, while $(a_i)$ measures the model’s pricing error for portfolio $i$.

Each coefficient captures a different dimension of risk:

- **$b_i$ (Market):** sensitivity to the excess market return $(R_M - R_f)$.
- **$s_i$ (Size / SMB):** exposure to the return spread between small and big firms.
- **$h_i$ (Value / HML):** exposure to the return spread between high and low book-to-market firms.
- **$r_i$ (Profitability / RMW):** sensitivity to the return spread between robust and weak profitability firms.
- **$c_i$ (Investment / CMA):** sensitivity to the return spread between conservative and aggressive investment firms.

Together, these loadings summarize how each portfolio co-moves with the factor premiums emphasized by Fama and French.


<h3 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.3em;">
Sample
</h3>

We use **monthly portfolio returns** and **monthly factor returns**, consistent with the original papers. All portfolio returns are converted to excess returns using the one-month Treasury bill rate. Our sample period matches the availability of the Fama–French Five-Factor dataset, and all portfolios are value-weighted as the author states in the data library.

The 25 portfolio sorts come from independent 5 x 5 sorts on Size and B/M, Size and OP, and Size and Inv. The 32 portfolio sorts come from 2 x 4 x 4 sorts on Size, B/M, and OP; Size, B/M, and Inv; and Size, OP, and Inv. These are the same portfolio definitions used in Fama and French (2015) to evaluate the three-factor and five-factor models.



<h3 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.3em;">
Estimation Technique
</h3>

To test whether all alphas are jointly zero within a given portfolio set (for example, all 25 Size–B/M portfolios), we use the **GRS test** of Gibbons, Ross, and Shanken (1989), following Fama and French (2014). The GRS statistic is computed for each model and each of the six portfolio families.

Regression parameters are estimated using Ordinary Least Squares (OLS), consistent with Fama and French (1993, 2015). Statistical inference centers on the significance of the factor coefficients and the intercepts (alphas), where a correctly specified model implies $a_i = 0$ for all test portfolios. We report conventional OLS $t$-statistics, matching the approach in the original studies.

We also evaluate:

- whether the estimated alphas $(a_i)$ differ significantly from zero,  
- the explanatory power of each regression (adjusted $R^2$),  
- whether the signs and magnitudes of the factor loadings align with economic intuition,  
- and the overall fit of the model across different portfolio sets.

<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
(d) Key Variables: Dependent and Independent Variables
</h2>

The dependent variable in all regressions is the **excess return on portfolio $i$ in month $t$**, defined as:

$$
Y_{it} = R_{it} - R_{Ft}.
$$

Here, $R_{it}$ is the value-weighted return on portfolio $i$ for month $t$, and $R_{Ft}$ is the one-month Treasury bill rate. Consistent with Fama and French (1993, 2015), all left hand side (LHS) portfolios are formed each June using NYSE breakpoints and their returns are computed monthly from July of year $t$ to June of year $t+1$. Value-weighted returns are computed as:

$$
R_{it} = \sum_{j \in \mathcal{P}_i} w_{j,t-1} R_{jt},
\qquad 
\sum_{j \in \mathcal{P}_i} w_{j,t-1} = 1.
$$

Our Y-portfolios follow the full Fama and French (2015) design. The test assets include:

- 25 Size–B/M portfolios  
- 25 Size–OP portfolios  
- 25 Size–Inv portfolios  
- 32 Size–B/M–OP portfolios  
- 32 Size–B/M–Inv portfolios  
- 32 Size–OP–Inv portfolios  

These portfolios produce large monotonic spreads across Size, Book to Market, Operating Profitability, and Investment. The key characteristics the models aim to explain.

<br>

<h3 style="font-family: Garamond; font-weight: bold; font-size: 18px; margin-bottom: 0.3em;">
Independent Variables (X): Factor Returns and Economic Interpretation
</h3>

The right hand side variables are the Fama–French factor returns. Each factor is a zero cost long–short portfolio intended to proxy for a systematic source of risk in expected returns. For each portfolio $i$, the estimated coefficients $(b_i, s_i, h_i, r_i, c_i)$ measure exposure to each factor.

<br>

**Market Factor (MKT):**  
$$
X_{1t} = R_{Mt} - R_{Ft}.
$$  
Economic interpretation: $b_i$ is the portfolio’s market beta, its sensitivity to aggregate economic risk.

<br>

**Size Factor (SMB):**  
$$
X_{2t} = SMB_t = R_{\text{Small},t} - R_{\text{Big},t}.
$$  
Economic interpretation: $s_i$ measures whether portfolio $i$ behaves like small firms ($s_i > 0$) or big firms ($s_i < 0$), reflecting the size premium.



<br>

**Value Factor (HML):**  
$$
X_{3t} = HML_t = R_{\text{High B/M},t} - R_{\text{Low B/M},t}.
$$  

HML measures the return premium investors earn for holding **value firms** rather than **growth firms**.

A **positive** $h_i$ means the portfolio behaves like value stocks and earns compensation for value-related risk.

A **negative** means the portfolio behaves more like growth stocks, which have lower expected returns.


<br>

**Profitability Factor (RMW):**  
$$
X_{4t} = RMW_t = R_{\text{Robust OP},t} - R_{\text{Weak OP},t}.
$$  

RMW measures the return premium associated with **profitability**, specifically the difference between firms with **robust operating profitability** and those with **weak profitability**.

A **positive** $r_i$ means the portfolio behaves like highly profitable firms and earns compensation for profitability related risk.

A **negative** means the portfolio behaves like weak profitability firms, which have lower and more volatile cash flows and therefore lower expected returns.


<br>

**Investment Factor (CMA):**  
$$
X_{5t} = CMA_t = R_{\text{Conservative Inv},t} - R_{\text{Aggressive Inv},t}.
$$  
Economic interpretation: $c_i$ reflects exposure to the investment premium. Conservatively investing firms earn higher expected returns than aggressively investing firms unless the latter have strong profitability.

<br>

Together, these factor loadings summarize how each portfolio co-moves with the underlying economic risks emphasized by Fama and French. A correctly specified factor model should absorb these systematic risks and leave intercepts that satisfy:

$$
a_i = 0 \quad \forall i.
$$


<h2 style="font-family: Garamond; font-weight: bold; font-size: 20px; margin-bottom: 0.3em;">
(e) Main Findings: Authors’ Primary Empirical Results and Conclusions
</h2>

The central empirical finding of Fama and French (2015) is that the Five-Factor Model offers a substantial improvement over the original Three-Factor Model. As they write, *“A five-factor model directed at capturing the size, value, profitability, and investment patterns in average stock returns performs better than the three-factor model of Fama and French (1993).”* The addition of profitability (RMW) and investment (CMA) factors sharply increases explanatory power and reduces the magnitude of pricing errors across most portfolio sets.

Despite this improvement, the model is not a perfect description of the data. The GRS test rejects the model as a complete asset-pricing model, even though its economic fit is strong. The authors note, *“The GRS test easily rejects a five-factor model directed at capturing these patterns, but we estimate that the model explains between 71% and 94% of the cross-section variance of expected returns for the Size, B/M, OP, and Inv portfolios we examine.”* In other words, the model is statistically rejected but economically powerful.

One of the most notable findings is that the profitability and investment factors render the original value factor (HML) redundant. The authors state directly that, *“With the addition of profitability and investment factors, the value factor of the FF three-factor model becomes redundant for describing average returns in the sample we examine.”* This redundancy arises because HML’s high average return can be almost entirely explained by its exposures to the other factors:  
*“…our results suggest that HML is a redundant factor in the sense that its high average return is fully captured by its exposures to $R_M - R_F$, SMB, and especially RMW and CMA.”*

This result highlights the theoretical success of the new factors:  
- **Profitability (RMW)** is strongly and positively related to average returns.  
- **Investment (CMA)** is strongly negatively related to average returns (firms investing aggressively earn lower returns).  

Together, these two factors subsume the information HML was capturing about cash-flow risk and investment behavior.

However, the Five-Factor Model also has a clear and well-defined failure. Fama and French describe this as the model’s *“lethal”* problem: its inability to explain the very low average returns on certain small, distressed firms. These are portfolios made up of small firms that **invest heavily despite having low profitability**, and they consistently generate pricing errors that the model cannot account for. As the authors write, *“Portfolios of small stocks with negative exposures to RMW and CMA are the biggest asset pricing problem in four of the six sets of LHS portfolios examined here.”* They conclude, *“…we find that the lethal portfolios contain small stocks of firms that invest a lot despite low profitability.”*

In summary, the main empirical outcomes are:

- The Five-Factor Model substantially improves upon the Three-Factor Model.  
- Profitability (RMW) is a powerful, economically meaningful factor with a strong positive relation to returns.  
- Investment (CMA) is also important, though it introduces challenges in capturing low-return, high-investment small firms.  
- The value factor (HML) becomes redundant once RMW and CMA are introduced.  
- Despite its successes, the model is still rejected statistically and fails most clearly for small, unprofitable, high-investment firms.

These findings form the basis of the paper’s contribution: a more complete but still imperfect model of the cross-section of expected returns.


## (f) Contribution: Explain what gap in the literature this paper addresses

___

# 2. Data Acquisition and Description

## (a) Data Sources

All data for this replication project comes from the **Kenneth R. French Data Library**, the public data archive maintained by the co-author of the Five-Factor Asset Pricing Model. The library provides the exact factor and portfolio files used in Fama and French (2015), already cleaned and formatted by the authors. Because the data is published directly by the original source, no external cleaning or reconstruction was required.

### Five-Factor Model Data
The five factor series—Mkt-RF, SMB, HML, RMW, CMA, and RF—was downloaded from the:

**Kenneth R. French Data Library**  
https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html

We use the **Changes in CRSP Data** monthly dataset, which contains the market excess return and the four additional factors introduced in the paper. The file covers data from July, 1963 onward and is updated regularly by the authors.

### Test Portfolio Data
To replicate the empirical setup in the original study, we also use the **`25 Portfolios Formed on Size and Book-to-Market (5x5)`** dataset from the same website. These portfolios are constructed using two-way sorts on firm size and book-to-market ratios and are the standard test assets used in evaluating factor models.

### Structure of the Author-Provided Files
Both datasets follow a consistent CSV format:

- Several descriptive header lines  
- A date column labeled "Unnamed: 0" containing YYYYMM date identifiers  
- Monthly percentage returns for each factor or portfolio  
- A copyright footer line  

Since these files are prepared by the authors themselves, they require almost no cleaning beyond removing the non-data header and footer lines. After these are removed, the files load correctly and can be merged using the date column.

## (a) Data Sources

All data for this replication project comes from the **Kenneth R. French Data Library**, the public data archive maintained by the co-author of the Five-Factor Asset Pricing Model. The library provides the exact factor and portfolio files used in Fama and French (2015), already cleaned and formatted by the authors. Because the data is published directly by the original source, no external cleaning or reconstruction was required.

### Five-Factor Model Data
The five factor series—Mkt-RF, SMB, HML, RMW, CMA, and RF—was downloaded from the:

**Kenneth R. French Data Library**  
https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html

We use the **Fama/French 5 Factors (2x3)** monthly dataset, which contains the market excess return and the four additional factors introduced in the paper. The file covers data from July 1963 onward and is updated regularly by the authors.

### Test Portfolio Data
To replicate the empirical setup in the original study, we also use the **`25 Portfolios Formed on Size and Book-to-Market (5x5)`** dataset from the same website. These portfolios are constructed using two-way sorts on firm size and book-to-market ratios and are the standard test assets used in evaluating factor models.

### How the Authors Source Their Underlying Data
While our replication uses the already-constructed CSV files published on French’s website, the **raw data** used by Fama and French in the original study comes from two institutional databases.  
Stock returns—including prices, total shares, delisting returns, and market equity—are obtained from the **CRSP (Center for Research in Security Prices)** database. Firm-level accounting variables such as book equity, operating profitability, and total assets are sourced from **Compustat**. The authors merge CRSP and Compustat data each June to form portfolios sorted by size, value, profitability, and investment. These portfolios are then used to compute the SMB, HML, RMW, and CMA factor returns. The factor and portfolio CSV files available on the Data Library are constructed using these procedures, which is why we do not need to rebuild the factors from raw CRSP or Compustat data ourselves.

### Structure of the Author-Provided Files
Both datasets follow a consistent CSV format:

- Several descriptive header lines  
- A date column labeled `"Unnamed: 0"` containing YYYYMM date identifiers  
- Monthly percentage returns for each factor or portfolio  
- A copyright footer line  

Since these files are prepared by the authors themselves, they require almost no cleaning beyond removing the non-data header and footer lines. After these are removed, the files load correctly and can be merged using the date column.


## (b) Data Processing

Because the datasets are published directly by Kenneth French, they require minimal cleaning. The main steps involved removing descriptive header and footer rows, renaming the date column, setting the date as the index, and merging the factor data with the portfolio returns so the regression could be estimated.

### Preparing the Five-Factor File
The five-factor CSV contains several descriptive header lines followed by a numerical dataset, along with a copyright footer at the bottom. These non-data rows were removed on import. The date column appears as "Unnamed: 0" in the raw file, so it was renamed to "date" and used as the index.

```python
fdata = pd.read_csv('fdata.csv', skiprows=4)
fdata = fdata[pd.to_numeric(fdata['Unnamed: 0'], errors='coerce').notna()]
fdata = fdata.rename(columns={'Unnamed: 0': 'date', 'Mkt-RF': 'MKTRF'})
fdata = fdata.set_index('date')
```
### Preparing the 25 Portfolios (5x5) File

The file containing the **25 Size–Book-to-Market portfolios (5x5)** is stored locally as **`data55.csv`**. This dataset includes 25 portfolio return series created from two-way sorts on firm size and book-to-market ratios. Similar to the factor file, it includes a long descriptive header and a non-numerical footer line.

These rows were removed by skipping the header on import and slicing off the final line. The date column was again labeled "Unnamed: 0" in the raw file and was renamed for consistency before setting it as the index.

```python
data55 = pd.read_csv('data55.csv', skiprows=15)
data55 = data55.rename(columns={'Unnamed: 0': 'date'})
data55 = data55.set_index('date')
data55 = data55[:-1]
```

This results in a clean monthly return series for all 25 portfolios, indexed by the same YYYYMM date format used in the factor dataset.

### Merging the Data

Once both datasets were cleaned and indexed by date, they were merged using an inner join. This ensures that each portfolio return observation is matched with the corresponding month’s factor values from the five-factor model.

```python
data = pd.merge(fdata, data55, on='date', how='inner')
```

The merged dataset contains the full set of portfolio returns along with the market, size, value, profitability, and investment factors. This final table is the basis for the regression analysis presented in Section 3.

### (c) Sample Description:

In [15]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
from statsmodels.regression.linear_model import OLS
import requests # This is the library for making HTTP requests. In this notebook, the module is grabbing data from Fama-French data library URL and downloading the .zip file. 
import zipfile # This is the built-in Python module for reading and writing ZIP archives. In this notebook,  after the requests library downloads the file, zipfile is used to open the .zip archive while it's still in memory. This allows us to extract the zip files.
from io import BytesIO, TextIOWrapper # This is a "translator" or "decoder." It takes a stream of raw bytes and turns it into a stream of readable text (strings). In this notebook, .txt filez inside the archive that is still just bytes, TextIOWrapper is used to "wrap" that byte stream, allowing us to read it as regular text.

In [11]:
fdata = pd.read_csv('factor_data.csv', skiprows=4)
fdata = fdata[pd.to_numeric(fdata['Unnamed: 0'], errors='coerce').notna()]
fdata = fdata.rename(columns = {'Unnamed: 0': 'date', 'Mkt-RF':'MKTRF'})
fdata = fdata.set_index('date')

# Converting data type to numeric
variables_to_convert = ['MKTRF', 'SMB', 'HML', 'RMW', 'CMA', 'RF']
fdata[variables_to_convert] = fdata[columns_to_convert].apply(pd.to_numeric, errors='coerce')

fdata

MKTRF    SMB    HML    RMW    CMA    RF
date                                           
196307  -0.39  -0.48  -0.81   0.64  -1.15  0.27
196308   5.08  -0.80   1.70   0.40  -0.38  0.25
196309  -1.57  -0.43   0.00  -0.78   0.15  0.27
196310   2.54  -1.34  -0.04   2.79  -2.25  0.29
196311  -0.86  -0.85   1.73  -0.43   2.27  0.27
...       ...    ...    ...    ...    ...   ...
  2020  23.65   3.74 -46.94  -5.21 -11.70  0.44
  2021  23.87  -1.02  25.61  26.72  11.67  0.04
  2022 -21.27  -1.57  25.68   6.55  22.50  1.42
  2023  21.75  -6.31 -13.98   6.24 -20.61  4.95
  2024  19.79 -13.52  -8.71   5.03  -9.83  5.26

[808 rows x 6 columns]

In [13]:
fdata.describe()

MKTRF         SMB         HML         RMW         CMA          RF
count  808.000000  808.000000  808.000000  808.000000  808.000000  808.000000
mean     1.123639    0.367475    0.532290    0.517401    0.459988    0.676584
std      6.713008    4.716616    5.219738    3.463540    3.541550    1.430856
min    -38.250000  -27.750000  -46.940000  -28.940000  -20.610000    0.000000
25%     -2.015000   -1.677500   -1.492500   -0.862500   -1.060000    0.160000
50%      1.090000    0.045000    0.230000    0.285000    0.120000    0.390000
75%      3.870000    2.212500    2.060000    1.510000    1.580000    0.540000
max     35.180000   50.180000   45.280000   26.720000   30.010000   14.710000

This analysis uses monthly returns for the Fama-French five factors: market excess return ($R_M−R_F$), size ($SMB$), value ($HML$), profitability ($RMW$), and investment ($CMA$), plus the risk-free rate ($R_F$). The descriptive statistics provided in the paper can be found at page 7 in table 4, panel A (2x3 Factors).

Our samples contain 808 monthly observations, which is significantly larger than the 606 monthly observations used in the original Fama and French (2015) paper. The paper's samples cover the period from July 1963 to December 2013. The 202-month difference is simply due to the use of a more recent dataset.

**Descriptive Statistics Comparison**

The table below compares the summary statistics from my sample (N=808) to those reported in the original paper (N=606). All values are presented as monthly percentages.

| Variable | Statistic | Our Samples (N=808) | Original Paper (N=606) |
| :--- | :--- | :--- | :--- |
| **$R_M-R_F$** | **Mean** | **1.1236** | **0.50** |
| | **Std dev.** | **6.7130** | **4.49** |
| **SMB** | **Mean** | **0.3675** | **0.29** |
| | **Std dev.** | **4.7166** | **3.07** |
| **HML** | **Mean** | **0.5323** | **0.37** |
| | **Std dev.** | **5.2197** | **2.88** |
| **RMW** | **Mean** | **0.5174** | **0.25** |
| | **Std dev.** | **3.4635** | **2.14** |
| **CMA** | **Mean** | **0.4599** | **0.33** |
| | **Std dev.** | **3.5416** | **2.01** |
| **RF** | **Mean** | **0.6766** | N/A |
| | **Std dev.** | **1.4309** | N/A |

**Analysis of Discrepancies**

The comparison reveals significant differences between the two samples, all of which can be attributed to the different sample periods.

Sample Size: As noted, my sample is 202 months longer. This extended period (post-December 2013) appears to have a substantial impact on the descriptive statistics.

**Means:** The mean for every single factor is higher in my sample than in the original paper. The differences are particularly stark for:

$R_M−R_F$ (Market): 1.12% vs. 0.50%.

$RMW$ (Profitability): 0.52% vs. 0.25%.

$HML$ (Value): 0.53% vs. 0.37%. 

**This suggests that the market, profitability, and value factors have all had very strong average performance in the period from 2014 to 2025.**

**Distributions (Standard Deviation):** The standard deviation (a proxy for volatility) is also dramatically higher across all factors in the samples.

The volatility of the market ($R_M−R_F$) is 6.71 vs. 4.49.

The volatility of $HML$ is 5.22 vs. 2.88.

The volatility of $SMB$ is 4.72 vs. 3.07.

**Potential Reasons:** The post-2013 data included in my sample covers several periods of extreme market volatility that are not in the paper's 1963-2013 sample. This includes the market shocks from the COVID-19 pandemic, the subsequent high-inflation environment, and the aggressive interest rate hike cycle. These events would explain both the higher average returns (driven by strong recoveries) and the much higher standard deviations seen in my sample's descriptive statistics.

The distributions from the samples, which also show large min and max values (e.g., -38.25% to 35.18% for MKTRF), confirm that the data includes periods of extreme market swings not fully captured in the original paper's 50-year sample.

### (d) Variable Construction:

* **Independent Variables (RHS Factors):**
    * **$R_M-R_F$**: The market excess return when compared to risk free asset return?


    * **SMB (Small Minus Big):** The size factor.

        * What it represents: This factor captures the size premium, which is the historical observation that small-company stocks (small market capitalization) tend to have higher average returns than large-company stocks.
     
            * How it's built:
              * Firms are sorted based on their market capitalization (Size).
              * The factor buys (goes long) a diversified portfolio of small stocks.
              * It sells (goes short) a diversified portfolio of big stocks.
                
          * What it means:
              * A positive SMB return means small-cap stocks outperformed large-cap stocks for that period.
              * A portfolio with a positive SMB beta is tilted toward small-cap stocks.


    * **HML (High Minus Low):** The value factor.

        * What it represents: This factor captures the value premium. This is the historical observation that "value" stocks (which appear cheap) tend to outperform "growth" stocks (which appear expensive).
          
         * How it's built:
             * Firms are sorted based on their Book-to-Market (B/M) ratio (the accounting value of the company divided by its stock market value).
             * A High B/M ratio means the stock is "cheap" (a value stock).
             * A Low B/M ratio means the stock is "expensive" (a growth stock).
             * The factor buys (goes long) a portfolio of High B/M (value) stocks.
             * It sells (goes short) a portfolio of Low B/M (growth) stocks.
           
        * What it means:
            * A positive HML return means value stocks outperformed growth stocks for that period.
            * A portfolio with a positive HML beta is tilted toward value stocks. A negative beta means it's tilted toward growth stocks.


    * **RMW (Robust Minus Weak):** The profitability factor.
 
        * What it represents: This factor captures the profitability premium. This is the observation that firms with high (robust) profitability tend to have higher average returns than firms with low (weak) profitability.
          
        * How it's built:
     
            * Firms are sorted based on their Operating Profitability (OP), which the paper measures as revenues minus costs, all divided by book equity.
            * The factor buys (goes long) a portfolio of Robust Profitability stocks.
            * It sells (goes short) a portfolio of Weak Profitability stocks.
              
        * What it means:
            * A positive RMW return means high-profitability firms outperformed low-profitability firms.
            * A portfolio with a positive RMW beta is tilted toward "quality" or high-profitability firms.

     
    * **CMA (Conservative Minus Aggressive):** The investment factor.

        * What it represents: This factor captures the investment premium. This is based on the theory and observation that companies that invest conservatively (grow their assets slowly) tend to have higher returns than firms that invest aggressively (grow their assets very quickly).
 
        * How it's built:
            * Firms are sorted based on their investment rate, which the paper measures as the year-over-year growth in total assets.
            * The factor buys (goes long) a portfolio of low-investment (Conservative) firms.
            * It sells (goes short) a portfolio of high-investment (Aggressive) firms.
              
        * What it means:
            * A positive CMA return means low-investment firms outperformed high-investment firms.
            * A portfolio with a positive CMA beta is tilted toward firms that are more conservative with their asset growth.

### (e) Missing Data:

Missing data is not a concern for this analysis. The descriptive statistics from our samples (N=808) show an identical count for all variables, indicating there are no missing observations (NaN values) within the loaded time-series dataset.

This is consistent with the methodology of the original paper. Fama and French (2015) do not impute missing data; rather, they handle it through sample selection. The samples are constructed by including "all NYSE, AMEX, and NASDAQ stocks on both CRSP and Compustat with...data for Size and B/M". Firms that are missing this required data are simply excluded from the sample before the portfolios are formed.

Because this analysis uses the pre-computed factor returns—which are the result of this rigorous screening process—no further steps were needed to handle missing data.

---

## 3. Replication Analysis

### (a) Model Specification: 

## (a) Model Specification

We estimate the Fama–French Five-Factor Model, which explains the excess return of a portfolio using five systematic risk factors. The regression equation is:

$$
R_{it} - R_{ft}
= \alpha_i
+ b_i \, (R_{Mt} - R_{ft})
+ s_i \, SMB_t
+ h_i \, HML_t
+ r_i \, RMW_t
+ c_i \, CMA_t
+ \varepsilon_{it}.
$$


where:

- **$R_{it} - R_{ft}$**  
  Portfolio *i*’s **excess return** at time *t*.  
  This is the dependent variable of the regression.

- **$\alpha$**  
  The portfolio’s **abnormal return** (intercept).  
  If the model prices the portfolio correctly, $\alpha_i$ should be statistically indistinguishable from zero.

- **$R_{Mt} - R_{ft}$** (MKT-RF)  
  The **market excess return**.  
  Captures the portfolio’s exposure to overall market risk.

- **$SMB_t$**  
  “Small Minus Big.”  
  The **size factor**, measuring the excess return of small firms over large firms.

- **$HML_t$**  
  “High Minus Low.”  
  The **value factor**, measuring the return spread between high book-to-market (value) and low book-to-market (growth) firms.

- **$RMW_t$**  
  “Robust Minus Weak.”  
  The **profitability factor**, capturing the return difference between firms with robust vs. weak operating profitability.

- **$CMA_t$**  
  “Conservative Minus Aggressive.”  
  The **investment factor**, measuring the return spread between firms that invest conservatively and those that invest aggressively.

- **$\varepsilon_{it}$**  
  Regression error term.

 **Coefficients**
- $b_i$: loading on the market factor  
- $s_i$: loading on the size factor  
- $h_i$: loading on the value factor  
- $r_i$: loading on the profitability factor  
- $c_i$: loading on the investment factor
  

These coefficients measure how sensitively the portfolio’s excess return responds to movements in each factor. The model is estimated using monthly data and Newey–West (HAC) standard errors to correct for potential autocorrelation and heteroskedasticity.


**Loading the Model:**

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import statsmodels.formula.api as smf
from scipy import stats

factors = pd.read_csv('5f_data.csv', skiprows=4)
factors = factors[pd.to_numeric(factors['Unnamed: 0'], errors='coerce').notna()]
factors = factors.rename(columns={'Unnamed: 0': 'date', 'Mkt-RF': 'MKTRF'})
factors = factors[['date', 'MKTRF', 'SMB', 'HML', 'RMW', 'CMA', 'RF']]
factors.iloc[:, 1:] = factors.iloc[:, 1:].apply(pd.to_numeric, errors='coerce')

port = pd.read_csv('25_portfolios_5x5.csv', skiprows=15)
port = port.rename(columns={'Unnamed: 0': 'date'})
port = port[pd.to_numeric(port['date'], errors='coerce').notna()]
port = port[:-1]
port.iloc[:, 1:] = port.iloc[:, 1:].apply(pd.to_numeric, errors='coerce')

factors = factors.drop_duplicates(subset='date')
port = port.drop_duplicates(subset='date')

data = pd.merge(port, factors, on='date', how='inner')

### (b) Estimation Results:

**Regression of Small, Low Book-to-Market Portfolios**

In [4]:
cols = ['SMALL LoBM', 'RF', 'MKTRF', 'SMB', 'HML', 'RMW', 'CMA']
data[cols] = data[cols].apply(pd.to_numeric, errors='coerce')

data['excess'] = data['SMALL LoBM'] - data['RF']
data['excess'] = pd.to_numeric(data['excess'], errors='coerce')

reg = smf.ols('excess ~ MKTRF + SMB + HML + RMW + CMA',
              data=data).fit(cov_type='HAC', cov_kwds={'maxlags':12})

reg.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 excess   R-squared:                       0.931
Model:                            OLS   Adj. R-squared:                  0.930
Method:                 Least Squares   F-statistic:                     1245.
Date:                Mon, 17 Nov 2025   Prob (F-statistic):               0.00
Time:                        22:11:19   Log-Likelihood:                -2100.8
No. Observations:                 808   AIC:                             4214.
Df Residuals:                     802   BIC:                             4242.
Df Model:                           5                                         
Covariance Type:                  HAC                                         
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.3973      0.102     -3.898      0.000      -0.597      -0.198
MKTRF          1.0255      0.019     54.576      0.000       0.989       1.062
SMB            1.4249      0.081     17.604      0.000       1.266       1.583
HML           -0.5236      0.066     -7.891      0.000      -0.654      -0.394
RMW           -0.4765      0.079     -5.999      0.000      -0.632      -0.321
CMA           -0.1862      0.102     -1.821      0.069      -0.387       0.014
==============================================================================
Omnibus:                      223.103   Durbin-Watson:                   2.177
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            11175.955
Skew:                           0.365   Prob(JB):                         0.00
Kurtosis:                      21.205   Cond. No.                         7.67
==============================================================================

Notes:
[1] Standard Errors are heteroscedasticity and autocorrelation robust (HAC) using 12 lags and without small sample correction
"""

In [5]:
factor_cols = ['MKTRF', 'SMB', 'HML', 'RMW', 'CMA', 'RF']
portfolio_cols = [c for c in data.columns if c not in factor_cols + ['date']]

results = []

for port in portfolio_cols:

    excess = pd.to_numeric(data[port], errors='coerce') - pd.to_numeric(data['RF'], errors='coerce')
    data['excess'] = excess


    reg = smf.ols('excess ~ MKTRF + SMB + HML + RMW + CMA',
                  data=data).fit(cov_type='HAC', cov_kwds={'maxlags':12})

    results.append({
        'Portfolio': port,
        'Alpha': reg.params['Intercept'],
        't(Alpha)': reg.tvalues['Intercept'],
        'b_MKT': reg.params['MKTRF'],
        't(MKT)': reg.tvalues['MKTRF'],
        'b_SMB': reg.params['SMB'],
        't(SMB)': reg.tvalues['SMB'],
        'b_HML': reg.params['HML'],
        't(HML)': reg.tvalues['HML'],
        'b_RMW': reg.params['RMW'],
        't(RMW)': reg.tvalues['RMW'],
        'b_CMA': reg.params['CMA'],
        't(CMA)': reg.tvalues['CMA'],
        'R2': reg.rsquared
    })


results_df = pd.DataFrame(results)
results_df = results_df.set_index('Portfolio')
results_df

Alpha  t(Alpha)     b_MKT     t(MKT)     b_SMB     t(SMB)  \
Portfolio                                                                  
SMALL LoBM -0.397253 -3.898092  1.025506  54.575718  1.424851  17.603811   
ME1 BM2     0.070876  0.847100  1.014304  41.213495  1.283305  23.559561   
ME1 BM3    -0.068010 -1.155964  0.963359  70.368098  1.145459  35.182687   
ME1 BM4     0.131476  2.180932  0.917889  37.767966  1.140933  20.818580   
SMALL HiBM  0.075392  0.778553  1.089643  19.796656  1.131575  30.804731   
ME2 BM1    -0.111067 -1.869746  1.043603  50.104368  0.973737  39.080564   
ME2 BM2     0.044552  0.607668  0.978379  42.056025  0.877067  31.410861   
ME2 BM3     0.016955  0.297575  0.958979  45.054777  0.841785  17.375956   
ME2 BM4     0.022421  0.419530  0.932719  33.282416  0.769374  31.541191   
ME2 BM5    -0.001795 -0.027849  1.035405  61.878573  0.812104  20.019199   
ME3 BM1     0.016899  0.326102  1.021216  42.937814  0.668573  18.312970   
ME3 BM2     0.028969  0.404408  0.990050  35.234114  0.626767  18.404762   
ME3 BM3    -0.085663 -1.162901  0.903090  32.127722  0.540565  16.681081   
ME3 BM4    -0.031091 -0.356206  0.994448  59.008265  0.558004  15.102074   
ME3 BM5     0.004814  0.075168  1.005230  36.401284  0.666899  17.950082   
ME4 BM1     0.198024  2.311670  1.021755  41.930619  0.384647   6.754000   
ME4 BM2    -0.204660 -2.141502  1.001512  25.912449  0.287466   4.358778   
ME4 BM3    -0.146943 -1.650690  0.988773  31.340611  0.260936   3.649176   
ME4 BM4     0.056704  0.905688  1.021827  42.187573  0.319074   6.830756   
ME4 BM5    -0.118395 -1.585820  1.121827  34.081362  0.333975   9.804290   
BIG LoBM    0.155538  2.594686  1.022733  44.608382 -0.212817  -5.273310   
ME5 BM2    -0.115612 -1.896578  0.954860  34.097641 -0.096253  -1.931755   
ME5 BM3    -0.124518 -1.563844  0.917002  38.442721 -0.110670  -2.277207   
ME5 BM4    -0.301604 -3.148061  0.961687  30.624061 -0.170825  -3.749776   
BIG HiBM    0.021390  0.246327  1.058775  36.354831 -0.167676  -4.429535   
excess     -0.494793 -3.946301  1.012803  25.695475 -0.189375  -3.823241   

               b_HML     t(HML)     b_RMW    t(RMW)     b_CMA    t(CMA)  \
Portfolio                                                                 
SMALL LoBM -0.523596  -7.890843 -0.476467 -5.999448 -0.186204 -1.820704   
ME1 BM2    -0.147326  -2.987263 -0.270765 -5.112507  0.061591  0.731814   
ME1 BM3     0.090504   3.613419 -0.048505 -1.532277  0.103139  1.978348   
ME1 BM4     0.293337   5.188148  0.074883  1.149166  0.132695  1.065794   
SMALL HiBM  0.428595   8.652273  0.106069  0.815717  0.356456  3.657099   
ME2 BM1    -0.471401 -13.023531 -0.202832 -5.519836 -0.093063 -1.316938   
ME2 BM2    -0.043596  -1.098721  0.040319  0.707390  0.008368  0.120608   
ME2 BM3     0.286700   8.987576  0.250838  5.378286 -0.113851 -1.825202   
ME2 BM4     0.478778  14.347005  0.159902  4.787891  0.002766  0.037662   
ME2 BM5     0.643564   7.976425  0.070092  1.483817  0.041525  0.565496   
ME3 BM1    -0.477713 -11.125325 -0.165869 -3.805939 -0.101479 -1.715709   
ME3 BM2     0.010919   0.150119  0.135719  3.146686  0.073707  1.114768   
ME3 BM3     0.349894   7.430880  0.200592  2.748873 -0.026633 -0.708039   
ME3 BM4     0.443980   5.743470  0.141531  1.947622  0.108191  1.487283   
ME3 BM5     0.634432  13.733782  0.106704  1.847950  0.200367  2.409550   
ME4 BM1    -0.514049 -12.112314 -0.067906 -1.327514  0.124279  1.644414   
ME4 BM2     0.025433   0.505079  0.171304  1.821945  0.202521  3.356885   
ME4 BM3     0.379484   6.240004  0.142491  1.940088  0.054911  0.951156   
ME4 BM4     0.487690   8.250782  0.063344  1.085053  0.028645  0.429594   
ME4 BM5     0.733533  13.960146 -0.040669 -0.796284 -0.022735 -0.299705   
BIG LoBM   -0.304947 -10.692267  0.248978  5.154548 -0.030243 -0.629034   
ME5 BM2     0.016254   0.532660  0.118692  3.932672  0.158164  2.824592   
ME5 BM3     0.170311   2.668030  0.058995  0.711644  0.197901  4.431241   
ME5 BM4    

### (c) Comparison with Original:

Create side-by-side comparison:

• Your coefficients vs. original paper’s coefficients

• Your standard errors vs. original standard errors

• Your R2 and other fit statistics vs. original

• Statistical significance patterns (which variables are significant in each)


### (d) Replication Assessment: Evaluate success of replication:

• Are coefficient signs the same?

• Are magnitudes similar? Calculate percentage differences

• Are significance levels comparable?

• Overall assessment: successful, partial, or unsuccessful replication


### (e) Discrepancy Investigation: If results differ, investigate potential reasons:

• Different sample periods or data versions

• Missing observations or variables

• Ambiguous specification details in original paper

• Software differences (R vs. Stata vs. Python)

• Errors in original paper or your implementation